components/cam/src/dynamics/se/inidat.F90

```diff
@@ -451,30 +452,45 @@ subroutine read_inidat( ncid_ini, ncid_topo, dyn_in)
           end do
     end do
 
+    read_pg_grid = .false.
     if ( (ideal_phys .or. aqua_planet)) then
        tmp(:,1,:) = 0._r8
     else    
       fieldname = 'PHIS'
       tmp(:,1,:) = 0.0_r8
-      if (fv_nphys > 0) then
+      read_pg_grid = fv_nphys > 0 .and. se_fv_phys_remap_alg == 0
+      if (read_pg_grid) then
          ! Copy phis field to GLL grid
          call infld(fieldname, ncid_topo, 'ncol', 1, nphys_sq, &
               1, nelemd, phys_tmp, found, gridname='physgrid_d')
-         if (se_fv_phys_remap_alg == 0) then
-            call fv_phys_to_dyn_topo(elem,phys_tmp)
+         call fv_phys_to_dyn_topo(elem,phys_tmp)
+      else
+         ! Attempt to read a mixed GLL-FV topo file, which contains PHIS_d in
```

Comment:
I think we need a better description of what happens when the topo is on the GLL grid. Maybe I'm behind on this discussion, but last time we talked about it I thought we were sure that the topo used by the dynamics needs to be defined on the FV grid for the purpose of the the parameterizations that handle the "sub-grid" topography that live in physics world. 
Is this change just a convenience so we don't have to create the topo on the FV grid? In other words, it allows us to use the conventional smoothing of topo on the GLL grid rather than defining a new smoothing operator on the FV grid?

Reply:
First, let me quote a commit message so we have this information available in the conversation:

I. Background. Topography data include phi_s, SGH, SGH30, and landm_coslat. SGH
and SGH30
    area(Omega)^-1 sqrt( integral_Omega (h1(x) - h2(x))^2 dx ),
where h1 and h2 are two topography references.

SGH captures the smoothing between an ~5-km length scale and the actual
grid. SGH30 captures the smoothing between the ~5-km length scale and the true
(30") topography.

1. We want phi_s on the dynamics grid and the FV grid to be consistent in the
sense that an integral over an FV subcell of either representation gives the
same value.

2. We also what topography to be as rough as the simulation can tolerate.

1 is currently true because we read FV PHIS, remap to GLL, then remap back to FV
to make 1 true. But this procedure does not satisfy 2 because it introduces new
smoothing.

II. New topo file format, the mixed GLL-FV format. Thus, a new topography file
format will have the fields
   PHIS_d, phi_s on the GLL grid
   PHIS, phis_s on the FV grid
   SGH, SGH30, LANDM_COSLAT, LANDFRAC on the FV grid
The old file format, which does not contain PHIS_d, is still supported.

III. Final tool chain. PHIS_d will have exactly the smoothness the dycore
requires. Then PHIS_d is remapped to PHIS. The other fields can be computed from
PHIS. This requires cube_to_target and other tools.

IV. Converter. This PR provides a tool in standalone Homme to convert an old GLL
file to a new mixed GLL-FV file. The remapped SGH is augmented with additional
variance due to the remap. The results are good, but the final tool chain should
be used for new grid development work.

Reply:
Does this mean the dual grid is no longer needed?

Reply:
I believe adoption of (1) physgrid and (2) tri-grid will make the GLL dual grid unnecessary in v2. This PR does not change the status of the dual grid; it was already unnecessary for physgrid topo. This PR focuses on maximizing the quality of topo on each grid while keeping the two topo representations consistent, which does not require the GLL dual grid.

SCREAM may not use tri-grid, in which case the GLL dual grid will still be needed for that project. But I believe the status of that point is uncertain.